pkg/controllers/volume-mgmt/handler.go

```diff
@@ -757,3 +758,35 @@ func (c *CStorVolumeController) reconcileVersion(cv *apis.CStorVolume) (*apis.CS
 	}
 	return cv, nil
 }
+
+// markCVStatusToOffline will fetch all the CV resources present
+// in etcd and mark it them CV.Status to Offline
+func (c *CStorVolumeController) markCVStatusToOffline() {
+	
+	cvList, err := c.clientset.CstorV1().CStorVolumes("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to fetch CV list, error: %v", err)
+		return
+	}
+	cvID := os.Getenv(string(common.OpenEBSIOCSPIID))
+	for _, cv := range cvList.Items {
+		cv := cv // pin it
+		if string(cv.GetUID()) == cvID {
+			// If target container restarts or pod is deleted
+			// before even creating a cvr, then we can skip those cvr
+			if cv.Status.Phase == "" || cv.Status.Phase == apis.CStorVolumePhase(apis.CVRStatusInit) {
+				continue
+			}
+			// No CstorVolume enum exist, hence wrapping cvr status
+			cv.Status.Phase = apis.CStorVolumePhase(apis.CVRStatusOffline)
+
+			// updating cv status to offline 
+			_, err = c.clientset.CstorV1().CStorVolumes(cv.Namespace).Update(context.TODO(), &cv, metav1.UpdateOptions{})
+			if err != nil {
+				klog.Errorf("failed to update CVR: %s status to %s", cv.Name, apis.CVRStatusOffline)
+				continue
```

Comment:
Updating an object could result in an error due to conflict errors(most common), in which case...I think an error should be returned here which implies , function signature should be modified to return an error  :
```suggestion
		return err
```
And then when you're calling it from the controller, you can check for the error and if the error exists, you can requeue the object.